### PR TITLE
pre-commit: Keep prettier version in sync; add eslint fixer

### PR DESCRIPTION
**Pull request by pokey**
- Partially addresses #640

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet
- [ ] File issue to revert patch working around https://github.com/pnpm/meta-updater/issues/10
- [ ] Check boxes in #640

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,8 @@
   },
   "pnpm": {
     "patchedDependencies": {
-      "@docusaurus/theme-search-algolia@2.3.1": "patches/@docusaurus__theme-search-algolia@2.3.1.patch"
+      "@docusaurus/theme-search-algolia@2.3.1": "patches/@docusaurus__theme-search-algolia@2.3.1.patch",
+      "@pnpm/meta-updater@0.2.2": "patches/@pnpm__meta-updater@0.2.2.patch"
```
Comment (pokey) [inline package.json:44]: Patching until https://github.com/pnpm/meta-updater/issues/10 is fixed

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -36,10 +36,34 @@ repos:
...
-    rev: "v2.7.1"
+  - repo: local
     hooks:
       - id: prettier
+        name: prettier
+        types: [text]
+        language: node
+        entry: prettier --write --list-different --ignore-unknown
+        additional_dependencies:
+          - prettier@2.7.1
```
Comment (auscompgeek) [inline .pre-commit-config.yaml:47]: Why not use the technique here? https://github.com/pre-commit/mirrors-prettier/issues/29#issuecomment-1332667344

**Comment by pokey** [comment]
Just realised that we could prob avoid all this nonsense by just doing pre-commit fixing in a GitHub action rather than in pre-commit.ci. Then we could just start with a pnpm install and run local scripts using `pnpm ...`

**Comment by auscompgeek** [comment]
Indeed. If you still want autofix commits, you can use pre-commit.ci lite: https://pre-commit.ci/lite.html

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -36,10 +36,34 @@ repos:
...
+        args: []
+  - repo: local
+    hooks:
+      - id: eslint
```
Comment (auscompgeek) [inline .pre-commit-config.yaml:51]: eslint should run before prettier, so prettier can format eslint's fixes without having to run pre-commit twice

**Comment by pokey** [comment]
Closing in favour of #1404

**Pull request closed without merging**